lib/features/give/bloc/give/give_bloc.dart

```diff
@@ -106,24 +103,134 @@ class GiveBloc extends Bloc<GiveEvent, GiveState> {
   ) async {
     emit(state.copyWith(status: GiveStatus.loading));
     try {
-      final organisation = await _getOrganisation(event.nameSpace);
-      final transactionList =
-          _createTransationList(event.nameSpace, event.userGUID);
-
-      await _givtRepository.submitGivts(
-        guid: event.userGUID,
-        body: {'donations': GivtTransaction.toJsonList(transactionList)},
+      await _processGivts(
+        namespace: event.nameSpace,
+        userGUID: event.userGUID,
+        emit: emit,
       );
-      emit(
-        state.copyWith(
-          status: GiveStatus.readyToGive,
-          organisation: organisation,
-          givtTransactions: transactionList,
+    } catch (e) {
+      log(e.toString());
+      emit(state.copyWith(status: GiveStatus.error));
+    }
+  }
+
+  FutureOr<void> _onBeaconScanned(
+    GiveBTBeaconScanned event,
+    Emitter<GiveState> emit,
+  ) async {
+    if (state.status == GiveStatus.processingBeaconData) {
+      return;
+    }
+    emit(state.copyWith(status: GiveStatus.processingBeaconData));
+    try {
+      final hex = StringBuffer();
+      for (final b in event.serviceData[event.serviceUUID]!) {
+        hex.write(sprintf('%02x', [b]));
+      }
+
+      final beaconData = hex.toString();
+      final contains = beaconData.contains('61f7ed01') ||
```

Comment:
I think it would be helpful for further maintenance to add some brief explanation of the parsing process, since such vars like 'contains' or 'a' are not very self-explanatory :)

Reply:
We have internal documentation for those variables